src/expertsystem/amplitude/sympy.py

```diff
@@ -0,0 +1,428 @@
+        return sum(coherent_intensities)
+
+    def __create_parameter_couplings(
+        self, graph_groups: List[List[StateTransitionGraph[ParticleWithSpin]]]
+    ) -> None:
+        for graph_group in graph_groups:
+            for graph in graph_group:
+                self.name_generator.register_amplitude_coefficient_name(graph)
+
+    def __generate_coherent_intensity(
+        self,
+        graph_group: List[StateTransitionGraph[ParticleWithSpin]],
+    ) -> sy.Symbol:
+        graph_group_label = _get_graph_group_unique_label(graph_group)
+        component_symbol = sy.Function(fR"I[{graph_group_label}]")(x)
+        amplitudes: List[sy.Expr] = list()
+        for graph in graph_group:
+            sequential_graphs = (
+                perform_external_edge_identical_particle_combinatorics(graph)
+            )
+            for seq_graph in sequential_graphs:
+                amplitudes.append(self.__generate_sequential_decay(seq_graph))
+        self.__model.components[component_symbol] = sum(
+            map(lambda a: Dagger(a) * a, amplitudes)
+        )
+        return component_symbol
+
+    def __generate_sequential_decay(
+        self, graph: StateTransitionGraph[ParticleWithSpin]
+    ) -> sy.Symbol:
+        partial_decays_symbols: List[sy.Symbol] = [
+            self._generate_partial_decay(graph, node_id)
+            for node_id in graph.nodes
+        ]
+        sequential_amplitudes = reduce(operator.mul, partial_decays_symbols)
+
+        component_symbol = sy.Function(
+            f"A[{self.name_generator.generate_unique_amplitude_name(graph)}]"
+        )(x)
+        coefficient = self.__generate_amplitude_coefficient(graph)
+        prefactor = self.__generate_amplitude_prefactor(graph)
+        expression = coefficient * sequential_amplitudes
+        if prefactor is not None:
+            expression = prefactor * expression
+        self.__model.components[component_symbol] = expression
+        return component_symbol
+
+    def _generate_partial_decay(
+        self, graph: StateTransitionGraph[ParticleWithSpin], node_id: int
+    ) -> sy.Symbol:
+        in_edge_ids = graph.get_edge_ids_ingoing_to_node(node_id)
+        out_edge_ids = graph.get_edge_ids_outgoing_from_node(node_id)
+        if len(in_edge_ids) != 1 or len(out_edge_ids) != 2:
+            raise ValueError(
+                f"Node {node_id} does not represent a 1-to-2 body decay!"
+            )
+
+        parent = _HelicityParticle(*graph.get_edge_props(in_edge_ids[0]))
+        children: List[_HelicityParticle] = list()
+        for out_edge_id in out_edge_ids:
+            edge_props = graph.get_edge_props(out_edge_id)
+            children.append(_HelicityParticle(*edge_props))
+
+        wigner_d = Wigner.d(
+            j=sy.Rational(parent.particle.spin, 2),
+            m=sy.Rational(parent.helicity, 2),
+            mp=sy.Rational(children[0].helicity - children[0].helicity, 2),
+            beta=x,
```

Comment:
The alpha Euler angle is also needed as input (assigned to phi later on)

I also do not get what `x` would be

Reply:
b0daf78
But `x` indeed still needs to be computed somehow: currently it's just the four-momentum data sample